packages/react-core/src/modules/core/hooks/useAuthenticatedKnockClient.ts

```diff
@@ -10,13 +22,25 @@ function useAuthenticatedKnockClient(
   const knockRef = React.useRef<Knock | null>();
 
   return useMemo(() => {
-    if (knockRef.current) knockRef.current.teardown();
+    const currentKnock = knockRef.current;
+
+    // If the userId and the userToken changes then just reauth
+    if (
+      currentKnock &&
+      currentKnock.isAuthenticated() &&
+      (currentKnock.userId !== userId || currentKnock.userToken !== userToken)
+    ) {
+      authenticateWithOptions(currentKnock, userId, userToken, options);
+      return currentKnock;
+    }
```

Comment:
This allows the client to reauthenticate when the user Id or userToken changes, without needing to reinit the entire Knock client